mlrun/datastore/utils.py

```diff
@@ -180,3 +181,55 @@ def get_kafka_brokers_from_dict(options: dict, pop=False) -> typing.Optional[str
             FutureWarning,
         )
     return kafka_bootstrap_servers
+
+
+def revert_list_filters_to_tuple(additional_filters):
+    tuple_filters = []
+    if not additional_filters:
+        return tuple_filters
+    validate_additional_filters(additional_filters=additional_filters)
+    for additional_filter in additional_filters:
+        tuple_filters.append(tuple(additional_filter))
+    return tuple_filters
+
+
+def validate_additional_filters(additional_filters):
+    if not additional_filters:
+        return
+    for filter_tuple in additional_filters:
+        if not filter_tuple:
+            continue
+        if not isinstance(filter_tuple, (list, tuple)):
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"mlrun supports additional_filters only as a list of tuples."
```

Comment:
This error message is duplicated in 3 places. It seems to me like more specific error messages are in order.

Reply:
It is just a nested type check, it is making sense that type error can have multiple appearances.

Reply:
To the extent that it makes sense, it should be extracted like the other error.